src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1647,6 +1748,37 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode)
 			TargetEntry *newTargetEntry = copyObject(originalTargetEntry);
 			newTargetEntry->expr = newExpression;
 
+			/*
+			 * Detect new targets of type Var and add it to group clause list.
+			 * This case is expected only if the target entry has aggregates and
+			 * it is inside a repartitioned subquery. We create group by entry
+			 * for each Var in target list. This code does not check if this
+			 * Var was already in the target list or in group by clauses.
+			 */
+			if (IsA(newExpression, Var) && walkerContext->createGroupByClause)
+			{
+				Var *column = (Var *) newExpression;
+				Oid opLt = InvalidOid;
```

Comment:
minor: we generally prefer to give more descriptive names such as `opLt` => `lessThanOperator` etc